File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        
        stack<string> st;
        string t;
        for(int i=0; i<s.size(); i++){
            if(s[i]!=' '){
                t.push_back(s[i]);
            }
            else if(s[i]==' '){
                if(t.size()>0){
                    st.push(t);
                    t.clear();
                }
            }
        }

        if(t.size()!=0){
            st.push(t);
        }

        string ans;
        while(!st.empty()){
            ans = ans+st.top()+' ';
            st.pop();
        }
        ans.pop_back();
        return ans;
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String temp = "";
        Stack<String> st = new Stack<>();

        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == ' '){
                if(temp.length()!=0){
                    st.push(temp);
                    temp="";
                }
            }
            else{
                temp = temp+s.charAt(i);
            }
        }
        if(temp.length()!=0){
            st.push(temp);
        }

        String ans = "";
        while(!st.isEmpty()){
            String t = st.peek();
            st.pop();
            
                if(ans.length()==0){
                    ans = t;
                }
                else{
                    ans = ans + ' ' + t;
                }
           
        }

        return ans;
    }
}